hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/GraphIndexTransaction.java

```diff
@@ -766,9 +811,15 @@ private ConditionQuery constructSearchQuery(ConditionQuery query,
                 if (key instanceof Id && indexFields.contains(key)) {
                     // This is an index field of search index
                     Id field = (Id) key;
-                    String propValue = elem.<String>getPropertyValue(field);
-                    String fvalue = (String) originQuery.userpropValue(field);
-                    if (this.matchSearchIndexWords(propValue, fvalue)) {
+                    assert elem != null;
+                    HugeProperty<?> property = elem.getProperty(field);
+                    Object value = property.value();
+                    // join collection data with white space
+                    String propValue = value instanceof Collection ?
+                                       StringUtils.join(value, " ") :
+                                       (String) value;
```

Comment:
add a method propertyValueToString(), make ensure value.toString() at line 231 can also be applied this strategy